File: app/src/main/java/com/otraempresa/android/example/AbstractRequestMerchantSettingsTask.java
package com.otraempresa.android.example;

import android.content.Context;
import android.os.AsyncTask;
import android.widget.Toast;

import com.kushkipagos.android.Kushki;
import com.kushkipagos.android.KushkiEnvironment;
import com.kushkipagos.android.KushkiException;
import com.kushkipagos.android.MerchantSettings;

abstract public class AbstractRequestMerchantSettingsTask extends AsyncTask<String, Void, MerchantSettings> {

    protected final Kushki kushkiMerchantSettings;
    private final Context context;

    AbstractRequestMerchantSettingsTask(Context context){
        this.context = context;
        kushkiMerchantSettings = new Kushki("10000002036955013614148494909956", "USD", KushkiEnvironment.TESTING);
    }

    @Override
    protected MerchantSettings doInBackground(String... params){
        try{
            return requestMerchantSettings();
        } catch (KushkiException kushkiException) {
            throw new RuntimeException(kushkiException);
        }
    }

    @Override
    protected void onPostExecute(MerchantSettings merchantSettingsInfo){
        if(merchantSettingsInfo.getMerchant_name().equals("")){
            showToast("Code" + merchantSettingsInfo.getCode() +"\nMessage: "+merchantSettingsInfo.getMessage());
        }else{
            showToast("Merchant name: "+merchantSettingsInfo.getMerchant_name()
                    +"\nCountry: "+merchantSettingsInfo.getCountry()
                    +"\nProcessor name: "+merchantSettingsInfo.getProcessor_name()
                    +"\nprodBaconKey: "+merchantSettingsInfo.getProdBaconKey()
                    +"\nsandboxBaconKey: "+merchantSettingsInfo.getSandboxBaconKey()
                    +"\nprocessors: "+merchantSettingsInfo.getProcessors());
        }

    }

    protected abstract MerchantSettings requestMerchantSettings() throws KushkiException;

    private void showToast(String text) {
        Toast toast = Toast.makeText(context, text, Toast.LENGTH_SHORT);
        toast.show();
    }

}

File: app/src/main/java/com/kushkipagos/kushkidemo/contracts/AbstractRequestMerchantSettingsTask.kt
package com.kushkipagos.kushkidemo.contracts

import android.content.Context
import android.widget.Toast
import com.kushkipagos.exceptions.KushkiException
import com.kushkipagos.kushki.Kushki
import com.kushkipagos.kushki.KushkiEnvironment
import com.kushkipagos.kushkidemo.helpers.CoroutineTask
import com.kushkipagos.models.MerchantSettings

abstract class AbstractRequestMerchantSettingsTask: CoroutineTask<String, Void, MerchantSettings> {
    protected var _kushki: Kushki
    private var _context: Context

    constructor(context: Context){
        _context = context
        _kushki = Kushki("c308a8af32114b8c97f8ba191149df9b", "USD", KushkiEnvironment.QA)
    }

    @Throws(KushkiException::class)
    abstract fun requestMerchantSettings() : MerchantSettings

    override fun doInBackground(vararg params: String?): MerchantSettings {
        try {
            return requestMerchantSettings();
        } catch (kushkiException: KushkiException) {
            throw RuntimeException(kushkiException);
        }
    }

    override fun onPostExecute(param: MerchantSettings?) {
        val merchantSettings = param!!
        if (merchantSettings.merchant_name == "") {
            showToast("Code: ${merchantSettings.code} Message: ${merchantSettings.message}");
        } else {
            showToast(
                """
                    Merchant name: ${merchantSettings.merchant_name}
                    Country: ${merchantSettings.country}
                    Processor name: ${merchantSettings.processor_name}
                    prodBaconKey: ${merchantSettings.prodBaconKey}
                    sandboxBaconKey: ${merchantSettings.sandboxBaconKey}
                    processors: ${merchantSettings.processors}
                    """.trimIndent()
            )
        }
    }

    fun showToast(text: String){
        val toast = Toast.makeText(_context, text, Toast.LENGTH_LONG)
        toast.show()
    }
}